libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFNetwork.h

```diff
@@ -32,30 +32,58 @@
 
 NS_SWIFT_NAME(NetworkManaging)
 @protocol SFNetworkSessionManaging
-- (nonnull NSURLSession *)ephemeralSession:(nonnull NSURLSessionConfiguration *)sessionConfig;
-- (nonnull NSURLSession *)backgroundSession:(nonnull NSURLSessionConfiguration *)sessionConfig;
+- (nonnull NSURLSession *)ephemeralSession:(nonnull NSURLSessionConfiguration *)sessionConfig SFSDK_DEPRECATED(7.3, 8.0, "Use sessionWithIdentifier:sessionConfiguration instead");
+- (nonnull NSURLSession *)backgroundSession:(nonnull NSURLSessionConfiguration *)sessionConfig SFSDK_DEPRECATED(7.3, 8.0, "Use sessionWithIdentifier:sessionConfiguration instead");
+- (nonnull NSURLSession *)sessionWithIdentifier:(nonnull NSString *)identifier sessionConfiguration:(nonnull NSURLSessionConfiguration *)configuration;
 @end
 
+extern NSString * __nonnull const kSFNetworkEphemeralSessionIdentifier NS_SWIFT_NAME(SFNetworkEphemeralSessionIdentifier);
+extern NSString * __nonnull const kSFNetworkBackgroundSessionIdentifier NS_SWIFT_NAME(SFNetworkBackgroundSessionIdentifier);
+
 NS_SWIFT_NAME(Network)
 @interface SFNetwork : NSObject
 
 typedef void (^SFDataResponseBlock) (NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) NS_SWIFT_NAME(DataResponseBlock);
 
 @property (nonatomic, readonly, strong, nonnull) NSURLSession *activeSession;
+@property (class, readonly, nonnull) NSDictionary *sharedSessions;
```

Comment:
Keeping this as a `property` has a lot of problems - it doesn't play well with the quasi-singleton approach and the class methods. You need an instance of the class to read and write to this dictionary. Instead, you should probably use a `static` dictionary to store the shared sessions. For example, look at `SFSDKSalesforceAnalyticsManager` or any of our other quasi-singleton classes that follow the same design. So, this would be something like:
```
static NSMutableDictionary *sharedSessions = nil;
```